### ISSUE TITLE
Layout/SpaceBeforeBrackets breaks code by removing brackets

### ISSUE THREAD

**Issue by thorsteneckel** [rubocop-hq/rubocop#9438]
Running `Layout/SpaceBeforeBrackets` on the https://github.com/zammad/zammad codebase breaks code on various locations. It wrongly detects a space before brackets and then removed the bracket resulting in broken code. See example below.

--------

## Expected behavior

While the code is not nice (yet) at these locations it should pass IMHO or fix it correctly/not break the code.

## Actual behavior

Breaks code by removing perfectly fine placed bracket.

## Steps to reproduce the problem

Run `rubocop -a` on a file containing just this line:

```ruby
collections[ Ticket::Article::Sender.to_app_model ] = []
```

Watch it to turn into:
```ruby
collections Ticket::Article::Sender.to_app_model ] = []
```

Or for more:
```
cd /tmp
git clone https://github.com/zammad/zammad.git
cd zammad
rvm use 2.6.6
bundle
bundle update rubocop-performance rubocop-rails rubocop-rspec rubocop
bundle exec rubocop app/controllers/sessions/collection_base.rb
# or: bundle exec rubocop --parallel --only Layout/SpaceBeforeBrackets
```
## RuboCop version

```
$ [bundle exec] rubocop -V
1.9.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.6.6 x86_64-darwin19)
  - rubocop-performance 1.9.2
  - rubocop-rails 2.9.1
  - rubocop-rspec 2.1.0
```
